cloudant-sync-datastore-core/src/main/java/com/cloudant/sync/internal/replication/PullStrategy.java

```diff
@@ -223,12 +221,8 @@ private void replicate()
         }
 
         this.state.documentCounter = 0;
-        for (this.state.batchCounter = 1; this.state.batchCounter < this.batchLimitPerRun; this
-                .state.batchCounter++) {
-
-            if (this.state.cancel) {
-                return;
-            }
+        while(!this.state.cancel) {
```

Comment:
This change to the `while` loop means that if the replication was cancelled, we'd get the logging after the loop indicating that the pull completed, which could prove misleading when debugging from logs.